src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -699,22 +699,23 @@ int UnixNativeCodeManager::IsInProlog(MethodInfo * pMethodInfo, PTR_VOID pvAddre
 #elif defined(TARGET_RISCV64)
 
 // store pair with signed offset
-// 0100 00xx xxxxxxxx xxxx xxxx xxxx xxxx
-#define STW_PAIR_BITS 0x04000000
-#define STW_PAIR_MASK 0xFC000000
+#define STW_PAIR_BITS 0x00003023
+#define STW_PAIR_MASK 0x0000707F
 
-// add fp, sp, x
-// addi fp, sp, x
-// 0000 0001 100x xxxx xxxx xxxx 0000 0000
-#define ADD_FP_SP_BITS 0x01C00000
-#define ADD_FP_SP_MASK 0xFFFFE000
-
-#define STW_PAIR_RS1_MASK 0xF80
-#define STW_PAIR_RS1_SP  0xF80
-#define STW_PAIR_RS1_FP  0xF00
-#define STW_PAIR_RS2_MASK 0xF00
-#define STW_PAIR_RS2_FP  0xF00
-#define STW_PAIR_RS2_RA  0xF40
+// add[i] fp, sp, x
+#define ADD_FP_SP_BITS 0x00010413
+#define ADD_FP_SP_MASK 0x000FFFFF
+
+// add[i] sp, sp, x
```

Comment:
```suggestion
// addi fp, sp, x
#define ADD_FP_SP_BITS 0x00010413
#define ADD_FP_SP_MASK 0x000FFFFF

// addi sp, sp, x
```
`add` is a different opcode. Do you need to cover `add/sub` (when large stack offset needs to be synthesized to a register)?

Reply:
Updated. For the question, I will defer to @filipnavara. My take is we can add it as necessary (when we have a repro showing the need).

Reply:
(I hate this function because it's fragile and duplicates some of the information that is in unwinding / GC information, but let's leave that aside for now.)

The function has three state return value - we are conclusively inside a prolog / we are conclusively outside of prolog / we don't know. We only require the "we are conclusively of prolog" state to be correct. The "we don't know" state is suboptimal because it may prevent hijacking and cause longer GC pauses.

In fact, we don't even really care about the whole prolog. We only care about the `ra`/`sp` saved on the stack and the unwinding information being correct enough to find those locations so we can hijack the memory location containing `ra` and override it to redirect to GC poll routine.

The current code seems to be good enough for simple prologs and also for prologs that establish frame and then probe stack for >PAGE_SIZE frames. It may not work well for frames between 1Kb and 4Kb depending on what code is emitted, but I think it would return "we don't know" state for those so it's not going to crash...